### PR TITLE
Adds list of regions instead of just the best

### PR DESCRIPTION
**Pull request by dulte**
Addressing #580

### CODE REVIEW

File: pyaerocom/region.py [hunk 1]
```diff
@@ -367,7 +367,7 @@ def find_closest_region_coord(lat, lon, regions=None):
     matches = get_regions_coord(lat, lon, regions)
 
     if len(matches) == 1:
-        return matches[0]
+        return matches[0], matches
```
Comment (avaldebe) [inline pyaerocom/region.py:370]: You're changing the type of the return from `str` to `tuple[str, list[str]]`.
I would create a new data structure for the result. Something based on `typing.NamedTuple`, like this:

``` python
from __future__ import annotations
[...]
from typing import NamedTuple
[...]

class ClosestRegion(NamedTuple):
    best: str
    matches: list[str]

def find_closest_region_coord(
    lat: float,
    lon: float,
    regions: dict[str, Region] | None = None
) -> ClosestRegion:
    [...]
    return ClosestRegion(best, matches)
```

This way you get attribute access to the `find_closest_region_coord` resuts, e.g.

``` python
reg = find_closest_region_coord(lon, lat, regions=regions)
# closest region ID
reg.best
# all regions applicable
reg.matches 
```

File: pyaerocom/region.py [hunk 2]
```diff
@@ -367,7 +367,7 @@ def find_closest_region_coord(lat, lon, regions=None):
...
 
     min_dist = 1e6
```
Comment (avaldebe) [inline pyaerocom/region.py:372]: this part could be replaced by

``` python
matches.sort(key=lambda id: regions[id].distance_to_center(lat, lon))
return ClosestRegion(matches[0], matches)
```

**Review by avaldebe — COMMENTED** [review]
I have a couple of suggestions in order to improve the maintainability following up.

File: pyaerocom/aeroval/coldatatojson_helpers.py [hunk 3]
```diff
@@ -628,8 +628,8 @@ def _init_site_coord_arrays(data):
 def _get_stat_regions(lats, lons, regions):
     regs = []
     for (lat, lon) in zip(lats, lons):
-        reg = find_closest_region_coord(lat, lon, regions=regions)
-        regs.append(reg)
+        best, reg = find_closest_region_coord(lat, lon, regions=regions)
+        regs.append([best, reg])
     return regs
```
[the remark this answers is not in the document]
Comment (avaldebe) [reply pyaerocom/aeroval/coldatatojson_helpers.py:633]: or using the data structure introduced on my next comment

``` python
def _get_stat_regions(
    lats: list[float], 
    lons: list[float],
    regions: dict[str, Region] | None,
) -> list[ClosestRegion]:
    assert len(lats) == len(lons), "miss match number of lat/lon"
    return [
        find_closest_region_coord(lat, lon, regions=regions)
        for lat, lon in zip(lats, lons)
    ]
```

File: pyaerocom/region.py [hunk 4]
```diff
@@ -343,8 +345,12 @@ def get_regions_coord(lat, lon, regions=None):
     return matches
 
 
-def find_closest_region_coord(lat, lon, regions=None):
-    """Find region that has it's center closest to input coordinate
+def find_closest_region_coord(
+    lat: float,
+    lon: float,
+    regions: Optional[dict] = None,
+) -> List[str]:
```
Comment (avaldebe) [inline pyaerocom/region.py:352]: here you could use the `|` style of annotations, introduced on Python 3.10:

``` python
from __future__ import annotations # the very first import
...
def find_closest_region_coord(
    lat: float,
    lon: float,
    regions: dict | None = None,
) -> list[str]:
    ...
```

File: pyaerocom/region.py [hunk 5]
```diff
@@ -358,23 +364,26 @@ def find_closest_region_coord(lat, lon, regions=None):
...
+    #     return matches[0], matches
+
+    # min_dist = 1e6
+    # best = None
+    # for match in matches:
+    #     region = regions[match]
+    #     dist = region.distance_to_center(lat, lon)
+    #     if dist < min_dist:
+    #         min_dist = dist
+    #         best = match
```
Comment (avaldebe) [inline pyaerocom/region.py:387]: please remove dead code

**Review by avaldebe — CHANGES_REQUESTED** [review]
small changes, mostly cleaning up commented code

**Comment by AugustinMortier** [comment]
I believe this is connected to this issue, but it appears that some regions are never returned in the list of regions, such as 'N America': e.g: [AeronetL1.5-d-od550aer_Column_IFS-CTRL-od550aer.json](https://gitlab.met.no/aeroval/cams2-82/-/raw/master/IFS/map/AeronetL1.5-d-od550aer_Column_IFS-CTRL-od550aer.json)

As a result, the minimap is not highlighting any station in the selected region in the mini map.
Screenshot from 2022-06-07 11-14-56

Other regions also affected by this issue are:
- Rus,Bel,Ukr
- Pacific,Aust,NZ,

My suspicion is that those regions are probably crossing the -180/180 degrees longitude and that is maybe not well taken into account when fetching the closest region.

**Comment by AugustinMortier** [comment]
In addition, it seems that the 'Oceans' region is added to almost all of the stations. This is not affecteing the statistics calculation though, as the number of stations shown in each region seems to be correct. So we should make sure that the regions in the region list is consistent with the regions used in the statistics calculation.

e.g: AERONET: 
- WORLD: 300 stations
- Oceans: 30 stations

Screenshot from 2022-06-07 11-27-49

Screenshot from 2022-06-07 11-28-15

**Pull request merged**